### PR TITLE
Install the operator-sdk CLI and check `sdk-generate` in CI

### PR DESCRIPTION
**Pull request by tvoran**
Adds `sdk-generate` to `make manifests`, to ensure the custom resource definitions in the base ClusterServiceVersion are up to date and are checked in the build-pre-checks CI job. Also adds `HCPAuth` and `HCPVaultSecretsApp` to `PROJECT` along with the generated parts in the ClusterServiceVersion base.

### CODE REVIEW

**Review by benashz — APPROVED** [review]
LGTM!

File: Makefile [hunk 1]
```diff
@@ -591,6 +592,23 @@ copywrite: ## Download copywrite locally if necessary.
 yq: ## Download yq locally if necessary.
 	@./hack/install_yq.sh
 
+.PHONY: operator-sdk
```
Comment (benashz) [inline Makefile:595]: Looks good! I wonder if we could maybe update it to like the `yq` make target. That will ensure that we have expected version of the sdk, as well as guard against any GH rate-limiters. Might be worth doing now.

File: PROJECT [hunk 2]
```diff
@@ -74,6 +74,24 @@ resources:
   kind: VaultTransit
   path: github.com/hashicorp/vault-secrets-operator/api/v1beta1
   version: v1beta1
+- api:
```
Comment (benashz) [inline PROJECT:77]: :+1:

**Review by benashz — COMMENTED** [review]
Looks good! I just had one comment/suggestion about fetching the operator-sdk.

File: Makefile [hunk 3]
```diff
@@ -507,20 +508,20 @@ set-image-ubi: kustomize copy-config ## Set the controller UBI image
...
 .PHONY: sdk-generate
-sdk-generate: copywrite
-	operator-sdk generate kustomize manifests -q
+sdk-generate: copywrite operator-sdk
+	$(OPERATOR_SDK) generate kustomize manifests -q
 	@$(COPYWRITE) headers &> /dev/null
 
 .PHONY: bundle
-bundle: manifests kustomize sdk-generate set-image-ubi yq ## Generate bundle manifests and metadata, then validate generated files.
+bundle: manifests kustomize sdk-generate set-image-ubi yq operator-sdk ## Generate bundle manifests and metadata, then validate generated files.
```
Comment (thyton) [inline Makefile:516]: This is out of my curiosity, not my actual code review :). 

I see `operator-sdk` is added to `sdk-generate`, and `sdk-generate` is added to `manifests`. I'm curious how `manifests` in `bundle` is executed differently that `sdk-generate` and `operator-sdk` remain in `bundle` (i.e why we don't include only `manifests` calling `sdk-generate` which also calls `operator-sdk`).

File: Makefile (hunk 1 above)
Comment (tvoran) [reply Makefile:595]: Sure, changed in 4ceb1e2

Comment (benashz) [reply Makefile:595]: Thanks!

**Comment by benashz** [comment]
LGTM!

**Pull request merged**